### PR TITLE
add helm linter github action

### PR DESCRIPTION
**Pull request by tony-schndr**
### What this PR does

Adds a Helm chart linter to CI checks.  The step to run the lint is conditional, running only if there are changes to any of the detected charts.

This uses default installed lint / schema configuration, if/when Charts become more defined we can update.
Jira: https://issues.redhat.com/browse/ARO-10414

Link to demo recording: 

### Special notes for your reviewer

Testing involved making a change to an existing chart and ensuring that the lint step actually ran and output was similar to what I experienced running ct locally.

### CODE REVIEW

File: .github/workflows/helm-lint.yml
Comment (mjlshen) [inline .github/workflows/helm-lint.yml:1]: Could we version these by SHA with the most specific version commented if possible? Like https://github.com/Azure/ARO-HCP/blob/main/.github/workflows/ci-go.yml

This lets dependabot keep these up-to-date without getting any unexpected updates

**Review by mjlshen — COMMENTED** [review]
functionally LGTM

File: .github/workflows/helm-lint.yml
Comment (tony-schndr) [reply .github/workflows/helm-lint.yml:1]: Yes, that would be ideal.  Where does the SHA come from?

Comment (mjlshen) [reply .github/workflows/helm-lint.yml:1]: Oh it's the git commit SHA that lines up with the tag

Comment (tony-schndr) [reply .github/workflows/helm-lint.yml:1]: I updated the pipeline I added here and other pipelines that were not using the sha.

**Review by geoberle — APPROVED** [review]
LGTM

**Pull request merged**